src/Canvas.elm

```diff
@@ -112,6 +115,33 @@ type DrawOp
     | Batch (List DrawOp)
 
 
+{-| `Style` specifies the style to apply as a `FillStyle` or a `StrokeStyle`
+-}
+type Style
+  = ColorStyle Color
+  | PatternStyle Canvas Repeat
+  | GradientStyle Gradient
+
+{-| Specifies the axis/axes along which to replicate a pattern. For use with the `PatternStyle` `Style`.
+-}
+type Repeat
+  = Repeat
+  | RepeatX
+  | RepeatY
+  | NoRepeat
+
+{-| A `Gradient` may be used to fill rectangles, lines, etc. The `LinearGradient` takes a starting `Point` for the gradient, an ending `Point` for the gradient, and the `ColorStop`s to apply. The `RadialGradient` takes the `Point` of the center of the starting circle, a `Float` representing its radius, the `Point` representing the center of the ending circle, a `Float` representing its radius, and the `ColorStop`s to apply.
+-}
+type Gradient
+    = LinearGradient Point Point (List ColorStop)
+    | RadialGradient Point Float Point Float (List ColorStop)
```

Comment:
What do you think about doing it like this?
```elm

type Style
    = Color Color
    | Pattern Canvas Repeat
    | LinearGradient Point Point (List ColorStop)
    | RadialGradient Point Float Point Float (List ColorStop)
```

Reply:
Yeah, I guess that's a bit cleaner, isn't it? I'll make the change and revert the examples.